libraries/standard/mqtt/include/mqtt.h

```diff
@@ -64,9 +64,9 @@ typedef uint32_t (* MQTTGetCurrentTimeFunc_t )( void );
  * @param[in] pPacketInfo Information on the type of incoming MQTT packet.
  * @param[in] pDeserializedInfo Deserialized information from incoming packet.
  */
-typedef void (* MQTTEventCallback_t )( MQTTContext_t * pContext,
-                                       MQTTPacketInfo_t * pPacketInfo,
-                                       MQTTDeserializedInfo_t * pDeserializedInfo );
+typedef void (* MQTTEventCallback_t )( struct MQTTContext * pContext,
+                                       struct MQTTPacketInfo * pPacketInfo,
```

Comment:
Would it be better to move the structure definitions along with their type definitions above these callbacks so that we can consistently use the same type everywhere?

Reply:
Then what about MQTTEventCallback_t definition in MQTTContext_t

Reply:
oh I see. That is an interesting problem. What is the doxygen issue if we forward declare the structure and typedef it?

Reply:
Then we cannot make references to the members of the struct in the documentation like `#MQTTContext_t.networkBuffer` .

Reply:
You might have tried this already, but I'm wondering if a `@struct` tag helps at all

Reply:
Yea tried